### ISSUE TITLE
Using JSON values as arguments

### ISSUE THREAD

**Issue by andrepm06** [garris/BackstopJS#817]
It is possible to use some of backstop.json values as arguments in puppeteer functions?

For example:

      "label": "Sign Confirmation",
      "url": "URL.COM",
      "referenceUrl": "HOM.URL.COM",
      "onReadyScript": "puppet/custom/sign-confirmation.js",
      "delay": 0,
      "hideSelectors": [],
      "selectorExpansion": true,
      "misMatchThreshold" : 0.5,
      "requireSameDimensions": true
 

> puppet/custom/sign-confirmation.js

```
 module.exports = async (page) => {

     await login(page)

     await page.goto(referenceUrl + \SIGNCONFIRMATION)

 };
```

**Comment by garris** [comment]
see this example -- you get the scenario object while your onReadyScript is running.

https://github.com/garris/BackstopJS/blob/master/capture/engine_scripts/puppet/onReady.js

**Comment by andrepm06** [comment]
Great! 

I was also looking for a way to define wether the execution is generating references or actually testing.

To do something like:

```
if (RUN_MODE == 'reference') {
        let url = scenario.referenceUrl
} else {
       let url = scenario.url
```

It is possible to do something similar currently?

**Comment by garris** [comment]
Yes, you can do this.  Your script does actually do get more parameters -- here is the signature...
`(page, scenario, viewport, isReference, browser )`
The 4th parameter is a Boolean -- true if the current run is a reference.
